website/addons/mendeley/model.py

```diff
@@ -133,12 +133,37 @@ def _citation_for_mendeley_document(self, document):
         :return Citation:
         """
         csl = {
-            'id': document.json.get('id'),
-            'type': document.json.get('type'),
+            'id': document.json.get('id')
         }
 
-        if document.title:
-            csl['title'] = document.title
+        CSL_TYPE_MAP = {
+            'book_section': 'chapter',
+            'case': 'legal_case',
+            'computer_program': 'article',
+            'conference_proceedings': 'paper-conference',
+            'encyclopedia_article': 'entry-encyclopedia',
+            'film': 'motion_picture',
+            'generic': 'article',
+            'hearing': 'speech',
+            'journal': 'article-journal',
+            'magazine_article': 'article-magazine',
+            'newspaper_article': 'article-newspaper',
+            'statute': 'legislation',
+            'television_broadcast': 'broadcast',
+            'web_page': 'webpage',
+            'working_paper': 'report'
+        }
+
+        csl_type = document.json.get('type')
+
+        if csl_type in CSL_TYPE_MAP:
+            csl['type'] = CSL_TYPE_MAP[csl_type]
```

Comment:
It is required.  I would think the default should be 'article' since that
is what 'generic' maps to, but I don't think there is a defined default.
On Jun 5, 2015 10:06 PM, "Steven Loria" <notifications@github.com> wrote:

> In website/addons/mendeley/model.py
> <https://github.com/CenterForOpenScience/osf.io/pull/2950#discussion_r31863457>
> :
>
> > +            'film': 'motion_picture',
> > +            'generic': 'article',
> > +            'hearing': 'speech',
> > +            'journal': 'article-journal',
> > +            'magazine_article': 'article-magazine',
> > +            'newspaper_article': 'article-newspaper',
> > +            'statute': 'legislation',
> > +            'television_broadcast': 'broadcast',
> > +            'web_page': 'webpage',
> > +            'working_paper': 'report'
> > +        }
> > +
> > +        csl_type = document.json.get('type')
> > +
> > +        if csl_type in CSL_TYPE_MAP:
> > +            csl['type'] = CSL_TYPE_MAP[csl_type]
>
> OK, thanks for the clarification. Is the type entry required? Does it
> need a default?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CenterForOpenScience/osf.io/pull/2950/files#r31863457>
> .
>